### ISSUE TITLE
Enhancement: [prefer-nullish-coalescing] recognize "||="

### ISSUE THREAD

**Issue by jakebailey** [typescript-eslint/typescript-eslint#5229]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/prefer-nullish-coalescing

### Description

In cases where an assignment is made to a nullish variable via `||=`, it'd be nice if `??=` was suggested instead.

### Fail

```typescript
let foo: Map<string, string> | undefined;

foo ||= new Map();

foo = foo || new Map(); // These are already handled in prefer-nullish-coalescing.
(foo || new Map()).get("key");
```

### Pass

```typescript
let foo: Map<string, string> | undefined;

foo ??= new Map();

foo = foo ?? new Map(); // These are already handled in prefer-nullish-coalescing.
(foo ?? new Map()).get("key");
```

**Comment by glen-84** [comment]
@JoshuaKGoldberg Should this be closed?

**Comment by Josh-Cena** [comment]
Right, fixed by #5234

**Comment by JoshuaKGoldberg** [comment]
Heh, thanks - so many PRs targeting `v6` that didn't close their issue...
